.evergreen/ensure-cmake.sh

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bash
+
+## This script ensures the presence of a CMake executable matching a specific
+## version. After sourcing this script, a variable "$CMAKE_EXE" will refer to
+## the CMake executable to be used.
+##
+## The following environment variables are defined by this script, and can also
+## be defined by a caller to control this script. All variables are optional
+## and have defaults for this file:
+##
+##  - CMAKE_VERSION: The CMake version to obtain
+##  - CMAKE_CACHES_ROOT: The directory where all obtained CMake versions will be cached
+##  - CMAKE_CACHE_DIR: The directory where we will expect/write the CMake that we obtain.
+##  - CMAKE_EXE: A CMake executable to use. If set, no obtaining logic will execute.
+##  - CTEST_EXE: The CTest executable for the CMAKE_EXE.
+
+
+. "$(dirname "${BASH_SOURCE[0]}")/setup-env.sh"
+
+# The version that we will attempt to obtain:
+: "${CMAKE_VERSION:=3.21.4}"
+
+# Root of all cached versions
+: "${CMAKE_CACHES_ROOT:="$BUILD_CACHE_DIR/cmake"}"
+# Cache directory for this particular version:
+: "${CMAKE_CACHE_DIR:="$CMAKE_CACHES_ROOT/$CMAKE_VERSION"}"
+
+# The executable that we want to use (can be overriden by an invoker)
+: "${CMAKE_EXE:="$CMAKE_CACHE_DIR/bin/cmake$EXE_SUFFIX"}"
+: "${CTEST_EXE:="${CMAKE_EXE%cmake*}ctest"}"
+
+# Downloads a prebuilt CMake binary:
+_download_cmake() {
+    # Directory where we will extract to (temporary)
+    declare extract_dir="$CMAKE_CACHE_DIR.tmp"
```

Comment:
I would suggest using `mktemp` to give a randomized temporary directory name.  It's probably not the case that we require the ability to run multiple concurrent builds out of the same directory tree, but I still see it as beneficial here.

Reply:
Beyond trying to avoid racing on creating the cached copy, I don't see the benefit of creating a unique temporary directory. Even if this were written to a temporary location, there could still be a race on placing the final cache results. I don't know of any portable shell-only method of inter-process synchronization here.